src/relay/backend/vm/compiler.cc

```diff
@@ -863,6 +864,8 @@ void VMCompiler::Compile(Module mod,
 
 Module VMCompiler::OptimizeModule(const Module& mod, const TargetsMap& targets) {
   Array<Pass> pass_seqs;
+  Array<tvm::Expr> entry_functions{tvm::Expr{"main"}};
+  pass_seqs.push_back(transform::RemoveUnusedFunctions(entry_functions));
```

Comment:
I wonder if we should run this pass at the end of optimization. For example, some lambda lifted functions will be no longer used after they are inlined.

Reply:
This is a good idea, we can run it twice begin and end. But I check `InlinePrimitives`, it doesn't seem to inline identity function in test `test_remove_all_prelude_functions_but_referenced_functions`. I can fix it later.